src/http/chain-registry/FaucetRegistry.test.ts

```diff
@@ -0,0 +1,180 @@
+import { HttpClient } from '../client/http'
+import { IFaucet, IHttpGETRequest, IHttpPOSTRequest, IRegisterFaucetRequest } from '../types'
+
+import { FaucetRegistry } from './FaucetRegistry'
+
+const mockHTTPClient = {
+  get: jest.fn(),
+  post: jest.fn()
+}
+
+jest.mock('../client/http', () => ({
+  HttpClient: jest.fn().mockImplementation(() => mockHTTPClient)
+}))
+
+const mockFaucet: IFaucet = {
+  name: 'testChain',
+  tenantID: 'tenantID',
+  uuid: 'uuid',
+  chainRule: 'chainUUID',
+  creditorAccount: '0xCreditorAddrAcc'
+}
+
+describe('FaucetRegistry', () => {
+  let faucetRegistry: FaucetRegistry
+
+  beforeAll(() => {
+    faucetRegistry = new FaucetRegistry('endpoint:8081')
+  })
+
+  describe('fetch', () => {
+    it('should fetch faucets successfully', async () => {
+      const req: IHttpPOSTRequest = {
+        path: '/faucets'
+      }
+
+      try {
+        mockHTTPClient.get.mockReturnValue(
```

Comment:
Everywhere in your tests where you write `mockReturnValue` of a `Promise`, please use `mockResolvedValueOnce` or `mockRejectedValueOnce`

Reply:
That is a good trick, thanks